core/src/main/java/com/novoda/merlin/MerlinsBeard.java

```diff
@@ -65,8 +93,12 @@ public boolean isConnectedToWifi() {
      * @return boolean true if a mobile network connection is available.
      */
     public boolean isConnectedToMobileNetwork() {
-        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_MOBILE);
-        return networkInfo != null && networkInfo.isConnected();
+        if (androidVersion.isMarshmallowOrHigher()) {
```

Comment:
This is basically the same logic as `isConnectedToWifi()` with the exception of the network type, maybe we can extract and deduplicate?